exercises/movr/users/data/users_database.sql

```diff
@@ -1,9 +1,11 @@
 CREATE DATABASE movr_users;
 
 CREATE TABLE movr_users.users (
+	id UUID NOT NULL DEFAULT gen_random_uuid(),
```

Comment:
Ah sorry @kikiya I wasn't clear on `city` col.  I definitely think `city` is important as a col for users. We're 100% in agreement on that.  

I just didn't understand why in a RBR table `city` is included in a compound key since `id` is an UUID and implicitly partitioned on the `crdb_region` col.  I may be incorrectly reading how RBR tables work but including `city` confuses me (I wonder if it will confuse students?) when I try to understand what "we mean when we say a row can be associated with a locality."  If I'm a customer student creating a RBR table from the ex I may think that I need a compound PK with `city` in order to get RBR working. But I don't think `city` is needed.

I'm gonna brain dump of how I think things work. LMK if I'm off base. I'm thinking the main points are that:

- the UUID makes `id` globally unique - thus when used in RBR table, we avoid a unique check's of all MR regions thus avoiding a negative impact write performance.
- making the table RBR means it gets a hidden `crdb_region` - when a write in a production MR cluster occurs, a function gets the gateway region & auto populates the `crdb_region` for that row.  Here's this in the docs (https://www.cockroachlabs.com/docs/stable/alter-table.html#add-constraints):
-- "If you do not set a home region for a row in a regional by row table, it defaults to the value returned by the built-in function gateway_region()."

--In the users_database.sql you INSERT the gataway stuff by setting the MR region in `crdb_region` for each row:
```INSERT INTO movr_users.users (email,city,last_name,first_name,phone_numbers,crdb_region)
VALUES
  ('in.felis.nulla@protonmail.edu','Zaria','Daugherty','Tamara', ARRAY['505-505-5050'],'us-east'),
  ('etiam.laoreet@aol.couk','San Rafael','Tucker','Kareem',ARRAY['505-505-5051'],'us-west'),
  ('nullam.ut@yahoo.com','Bremerhaven','Nolan','Seth',ARRAY['505-505-5052'],'us-east'),
```
So in sum, in a production MR cluster `crdb_region` is auto populated. In the exercise, we explicitly set `crdb_region` when inserting each test row. Either way, each row has a locality. That seems like the important stuff we need students to understand. I'm thinking linking `city` to this may give students the impression it's needed for RBR locality when it's not. Am I off base here?

Reply:
Hi @dweatherbee, I only kept the `city` column. I removed the compound key as requested. 

Linking the `city` reinforces the idea that the RBR locality should be used when the data in a row can be directly associated with some locality _and that locality can be different for each row_.  It's not just important to know that each row has locality; it's important to understand how it's determined and why it makes sense to partition the data at the row level. Not all data makes sense to be partitioned at the row level. 

And yes, in production you can automatically populate the `crdb_region` column or explicitly set the region. The region could also be set using a computed value, which in the docs example, we use the `city` value to compute the region. (We also mention this method in the MR course)

`ALTER TABLE rides ADD COLUMN region crdb_internal_region AS (
  CASE
    WHEN city IN ('new york', 'boston', 'washington dc', 'chicago', 'detroit', 'minneapolis') THEN 'us-east-1'
    WHEN city IN ('san francisco', 'seattle', 'los angeles') THEN 'us-west-1'
    WHEN city IN ('amsterdam', 'paris', 'rome') THEN 'eu-west-1'
  END
) STORED;`

Again, understanding why you've determined data would benefit from row-level partitioning is key to how you decide to populate that `crdb_region` column. 

As to why docs uses a compound key with `city`, there could be many reason. I'm only speculating why ... With `city` in the pk the data would be stored close to each other on disk, reducing latency. Also, queries that sort or filter by `city` would be faster with `city` in the primary key. `city` affects the physical ordering of data on disk, `crdb_region` controls the node. It's used as yet another level of locality control. 

In the last commit, I completely removed the compound key. I initially included it to be consistent with docs, but it's definitely gone now.  Let me know if you'd like additional changes.

Reply:
Thx @kikiya this explanation is very helpful. And thx for removing the compound key.